src/SAML2/XML/AbstractConvertable.php

```diff
@@ -27,6 +29,36 @@ public function __toString(): string
     }
 
 
+    /**
+     * Serialize this XML chunk
+     *
+     * @return string The serialized chunk.
+     */
+    public function serialize(): string
+    {
+        return serialize($this->toXML()->ownerDocument->saveXML());
```

Comment:
There's no need to call `serialize()` here. The `serialize()` method of the `Serializable` interface needs to return a string that you can later recover, but that doesn't mean it needs to follow any particular format. Returning the XML document as a string should suffice, as we can recover it later from that representation.

Reply:
Done